# 00 Scope and Ethics

Notebook ini menyiapkan struktur proyek, memuat konfigurasi crawl, dan menegaskan guardrail pengumpulan data publik-only.

In [1]:
from pathlib import Path
import sys

PROJECT_ROOT = Path.cwd().resolve().parent if Path.cwd().name == 'notebooks' else Path.cwd().resolve()
SRC_ROOT = PROJECT_ROOT / 'src'
if str(SRC_ROOT) not in sys.path:
    sys.path.insert(0, str(SRC_ROOT))

from shopee_safe_scraper.config import DEFAULT_CONFIG, load_config, save_config
from shopee_safe_scraper.pipeline import ensure_project_layout

paths = ensure_project_layout(PROJECT_ROOT)
config_path = PROJECT_ROOT / 'config.yaml'
if not config_path.exists():
    save_config(config_path, DEFAULT_CONFIG)
config = load_config(config_path)
config

{'categories': ['Elektronik',
  'Fashion',
  'Kesehatan',
  'Kecantikan',
  'Rumah Tangga',
  'Olahraga'],
 'target_products_per_category': 9,
 'max_reviews_per_product': 100,
 'min_delay_s': 4,
 'max_delay_s': 9,
 'cooldown_every_n_requests': 10,
 'cooldown_min_s': 60,
 'cooldown_max_s': 120,
 'max_requests_per_run': 150,
 'timeout_s': 30,
 'stop_on_status': [403, 429],
 'stop_on_keywords': ['captcha',
  'verify you are human',
  'login',
  'sign in',
  'forbidden',
  'access denied'],
 'user_agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0 Safari/537.36',
 'public_only': True,
 'single_thread_only': True}

## Guardrails

- Hanya halaman publik.
- Tidak login, tidak bypass, tidak proxy rotation.
- Stop pada `403`, `429`, captcha, atau forced login.
- Manual seeding dipakai untuk membatasi scope dan risiko.